lib/ansible/parsing/dataloader.py

```diff
@@ -171,9 +171,16 @@ def set_basedir(self, basedir):
             self._basedir = to_text(basedir)
 
     def path_dwim(self, given):
-        '''
-        make relative paths work like folks expect.
-        '''
+        """Make relative paths work like folks expect.
+
+        :parama given: The path
+        :type data: str
+
+        :return: The modify path
+        :rtype: str
+        """
+        if not isinstance(given, string_types):
+            return given
```

Comment:
In what case should a path not be a string?  I question the validity of just returning what was given, if it wasn't a string.  I might instead expect this to raise a `TypeError` instead.